src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int tableLength;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        tableLength = INITIAL_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(getKeyHash(key), key, value, null);
+        size = putNodeAtPosition(newNode, getKeyHash(key) % tableLength) ? size + 1 : size;
+        if (size > tableLength * LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int position = getKeyHash(key) % tableLength;
+        Node<K, V> currentNode = table[position];
+        if (currentNode == null) {
+            return null;
+        }
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getKeyHash(K key) {
+        int hash = key == null ? 0 : key.hashCode();
+        return hash < 0 ? hash * -1 : hash;
+    }
+
+    private void resize() {
+        tableLength *= CAPACITY_MULTIPLIER;
+        Node<K, V>[] oldTable = table;
+        table = new Node[tableLength];
+        for (Node<K, V> kvNode : oldTable) {
+            if (kvNode != null) {
+                addAllNodeByHead(kvNode);
```

Comment:
you can call here put() method

Reply:
I have to remove all "next" links in each nodes from LinckedList in the bucket. So, if I call the put method here without removing the reference, then in the future, I will have problems. And i have to put not only LinckedLists`s head